core/lib.rs

```diff
@@ -294,18 +295,17 @@ impl Connection {
         self.pager.clear_page_cache();
         Ok(())
     }
-}
 
-impl Drop for Connection {
-    fn drop(&mut self) {
+    /// Close a connection and checkpoint.
```

Comment:
out of interest, why did this change

Reply:
is it just because the conns in the simulator dont end up being dropped so checkpointing doesnt happen via destructor

Reply:
It was because inject fault would inject a fault on drop checkpoints. I cannot capture these errors on a drop after its lifetime so I need control on when checkpoint/drop happens.